File: Objective-C/SNChart/SNChart/ViewController.m
//
//  ViewController.m
//  SNChart
//
//  Created by solon on 16/5/2.
//  Copyright © 2016年 solon. All rights reserved.
//

#import "ViewController.h"
#import "SNChartView.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
    SNChartView *chart = [[SNChartView alloc]init];
    chart.frame = CGRectMake(20, 100, 250, 300);
    NSValue *point1 = [NSValue valueWithCGPoint:CGPointMake(10, 150)];
    NSValue *point2 = [NSValue valueWithCGPoint:CGPointMake(50, 10)];
    NSValue *point3 = [NSValue valueWithCGPoint:CGPointMake(100, 200)];
    NSValue *point4 = [NSValue valueWithCGPoint:CGPointMake(200, 250)];

    chart.points = @[point1,point2,point3,point4];
    [self.view addSubview:chart];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift/LearniOS30Days/SNStopWatch/SNStopWatch/ViewController.swift
//
//  ViewController.swift
//  SNStopWatch
//
//  Created by solon on 16/10/29.
//  Copyright © 2016年 solon. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var counter = 0.0
    var timer = NSTimer()
    var isPlaying = false
    
    @IBOutlet weak var playBtn: UIButton!
    @IBOutlet weak var pauseBtn: UIButton!
    @IBOutlet weak var timeLabel: UILabel!

    
    @IBAction func resetAction(sender: AnyObject) {
        timer.invalidate()
        isPlaying = false;
        counter = 0.0
        timeLabel.text = String(counter)
        playBtn.enabled = true;
        pauseBtn.enabled = true;
    }
    
    @IBAction func playAction(sender: AnyObject) {
        if(isPlaying == true) {
            return;
        }
        playBtn.enabled = false;
        pauseBtn.enabled = true;
        timer = NSTimer.scheduledTimerWithTimeInterval(0.1, target: self, selector: Selector("updateTimer"), userInfo: nil, repeats: true)
        isPlaying = true;
    }
    
    @IBAction func pauseAction(sender: AnyObject) {
        playBtn.enabled = true;
        pauseBtn.enabled = false;
        timer.invalidate()
        isPlaying = false;
    }
    
    
    override func preferredStatusBarStyle() -> UIStatusBarStyle {
        return UIStatusBarStyle.LightContent
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        timeLabel.text = String(counter)
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func updateTimer(){
        counter = counter + 0.1
        timeLabel.text = String(format: "%.1f", counter)
    }
}

